charts/consul/values.yaml

```diff
@@ -86,7 +86,7 @@ global:
   # image that is used for functionality such as catalog sync.
   # This can be overridden per component.
   # @default: hashicorp/consul-k8s-control-plane:<latest version>
-  imageK8S: docker.mirror.hashicorp.services/hashicorppreview/consul-k8s-control-plane:1.4-dev
+  imageK8S: docker.mirror.hashicorp.services/hashicorppreview/consul-k8s-control-plane:1.4.2-dev
```

Comment:
I think we want to keep this at the floating tag similar to [`main`](https://github.com/hashicorp/consul-k8s/blob/main/charts/consul/values.yaml#L89) (same as above comment)